asv/commands/run.py

```diff
@@ -194,6 +196,17 @@ def run(cls, conf, range_spec=None, steps=None, bench=None, attribute=None, para
         elif range_spec == "ALL":
             # All commits on each configured branches
             commit_hashes = repo.get_new_branch_commits(conf.branches, [])
+        elif isinstance(range_spec, six.string_types) and range_spec.startswith('HASHFILE:'):
+            hashfn = range_spec[9:]
+            if hashfn == '-' or hashfn.lower() == 'stdin':
+                hashstr = sys.stdin.read()
+            elif os.path.isfile(hashfn):
+                with open(hashfn, 'r') as f:
+                    hashstr = f.read()
+            else:
+                log.error('Requested commit hash file "{}" is not a file'.format(hashfn))
+                return 1
+            commit_hashes = hashstr.strip().split('\n')
```

Comment:
`commit_hashes = [h.strip() for h in hashstr.split("\n") if h.strip()]`
To allow empty lines etc.